tests/functional/test_install_reqs.py

```diff
@@ -817,3 +817,50 @@ def test_config_settings_local_to_package(
     assert "--verbose" not in simple3_args
     simple2_args = simple2_sdist.args()
     assert "--verbose" not in simple2_args
+
+
+def test_install_options_no_deps(
+    script: PipTestEnvironment, resolver_variant: ResolverVariant
+) -> None:
+    create_basic_wheel_for_package(
+        script, "A", "0.1.0", depends=["B==0.1.0"], extras={"C": ["C"]}
+    )
+    create_basic_wheel_for_package(script, "B", "0.1.0")
+    create_basic_wheel_for_package(script, "C", "0.1.0")
+
+    requirements_txt = script.scratch_path / "requirements.txt"
+    requirements_txt.write_text("A[C] --no-deps")
```

Comment:
Since this is intended to be a per-requirement flag, I think we should add a test ensuring that it doesn't set a global context and have additional requirements with dependencies in a test like this.